src/commands/sfpowerkit/package/valid.ts

```diff
@@ -259,6 +253,59 @@ Elements supported included in your package testPackage are
 
     return sfdx_package;
   }
+  private setCoverageJSON(): void {
+    if (this.flags.version) {
+      try {
+        this.coverageJSON = this.findCoverageJSON();
+      } catch (fileError) {
+        this.coverageJSON = this.getDefaultCoverageJSON();
+        this.ux.log(
+          "Could not find specified metadata version in resources...\n" +
+            `Using default version ${this.coverageJSON.versions.selected}`
+        );
+      }
+    } else {
+      this.coverageJSON = this.getDefaultCoverageJSON();
+    }
+  }
+
+  private findCoverageJSON() {
+    let resourcesDir = path.join(
+      __dirname,
+      "..",
+      "..",
+      "..",
+      "..",
+      "resources"
+    );
+    const resources: string[] = fs.readdirSync(resourcesDir);
+    for (let i = 0; i < resources.length; i++) {
+      if (resources[i].match(/.json/i)) {
+        let resourcePath = path.join(resourcesDir, resources[i]);
```

Comment:
@aly76  Why do we need to iterate through the directory? As we know the api version, we just goto the resoruces section and read the file directly.. If its not there throw an error.